File: Python/242.Valid Anagram.py
#判断两个字符串是否是“相同字符异序词”
class Solution(object):
    def isAnagram(self, s, t):
        if len(s)!=len(t):  return False
        dic1 , dic2 = {}, {}
        for ch in s:
            if dic1.get(ch,-1)==-1:
                dic1[ch] = 1
            else:
                dic1[ch] += 1
        for ch in t:
            if dic2.get(ch,-1)==-1:
                dic2[ch] = 1
            else:
                dic2[ch] += 1
        return dic1==dic2

s = Solution()
print(s.isAnagram("anagram","nagaram"))

File: C++/Hash Table/242.Valid Anagram.cpp
/*
判断两个字符串是否是“相同字符异序词”
*/
#include <iostream>
#include <vector>
#include <unordered_map>
using namespace std;

class Solution {
public:
	bool isAnagram(string s, string t) {
        unordered_map<char,int>		mymap1,mymap2;
        for(int i=0; i < s.size(); ++i)
        {
        	if(mymap1.count(s[i])==0)
        		mymap1[s[i]]=0;
        	else
        		++mymap1[s[i]];
        	if(mymap2.count(t[i])==0)
        		mymap2[t[i]]=0;
        	else
        		++mymap2[t[i]];
        }
        return mymap1==mymap2 && s.size()==t.size();
    }
};

int main()
{
	Solution s;
	cout << s.isAnagram("anagram","nagaram") << endl;
}

